src/shipchain_common/iot.py

```diff
@@ -47,92 +36,19 @@ def __init__(self):
             aws_service='execute-api'
         )
 
+        self.url = f'https://{settings.IOT_AWS_HOST}/{settings.IOT_GATEWAY_STAGE}/'
         self.session = requests.session()
         self.session.headers = {'content-type': 'application/json'}
         self.session.auth = aws_auth
 
-    def _call(self, http_method, endpoint, payload=None, params=None):
-        generic_endpoint = AWSIoTClient._get_generic_endpoint_for_metric(http_method, endpoint)
-
-        if payload:
-            payload = json.dumps(payload, cls=DecimalEncoder)
-
-        url = f'https://{settings.IOT_AWS_HOST}/{settings.IOT_GATEWAY_STAGE}/{endpoint}'
-
-        try:
-
-            with TimingMetric('python_common_aws_iot.call', tags={'method': generic_endpoint}) as timer:
-
-                if http_method == AWSIoTClient.METHOD_POST:
-                    response = self.session.post(url, data=payload, params=params)
-                    response_json = response.json()
-
-                    if response.status_code != status.HTTP_201_CREATED:
-                        self._process_error_object(generic_endpoint, response, response_json)
-
-                elif http_method in AWSIoTClient.RESPONSE_200_METHODS:
-                    response = getattr(self.session, http_method)(url, data=payload, params=params)
-                    response_json = response.json()
-
-                    if response.status_code != status.HTTP_200_OK:
-                        self._process_error_object(generic_endpoint, response, response_json)
-
-                else:
-                    log_metric('python_common_aws_iot.error', tags={'method': generic_endpoint,
-                                                                   'code': 'InvalidHTTPMethod'})
-                    LOG.error('aws_iot_client(%s) error: %s', generic_endpoint, 'Invalid HTTP Method')
-                    raise AWSIoTError(f'Invalid HTTP Method {http_method}')
-
-                LOG.info('aws_iot_client(%s) duration: %.3f', generic_endpoint, timer.elapsed)
+    def _call(self, http_method, endpoint, metric_name=None, payload=None, params=None):
+        metric_name = AWSIoTClient._get_generic_endpoint_for_metric(http_method, endpoint)
```

Comment:
This caused me to think a lot trying to figure out the best way to have the metric_name, I still believe I could've thought of a better way. I feel like we should always require a metric_name, but AWSIoTClient was creating a generic one in the ._call that wouldn't have worked for the URLShortenerClient. Maybe it would be better to remove it as a parameter and instead make a function for each class that would make a generic metric_name?

Reply:
Yes you could do that. But if you want to go that route, I think it will be easier to just modify the behavior of `_get_generic_endpoint_for_metric`  to handle your case.

Reply:
Yeah that's what I was thinking. What gave me pause was that the endpoint was static and I wasn't sure if it was being used anywhere by profiles/transmission and I didn't want to break anything, which doesn't seem to be the case.